src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -336,6 +376,35 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
                 if (!calculateNextItinerary()) return true;
             }
             LOG.info("Next itinerary happening on {}.", targetDate);
+
+            // update journey state with baseline departure and arrival times which are the last known departure/arrival
+            journeyState.baselineDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+            journeyState.baselineArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+
+            // update journey state with the original (scheduled departure and arrival times). Calculate these by
+            // finding the first/last transit legs and subtracting any delay.
+            journeyState.originalDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+            for (Leg leg : matchingItinerary.legs) {
+                if (leg.transitLeg) {
+                    journeyState.originalDepartureTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                        leg.departureDelay,
+                        TimeUnit.SECONDS
+                    );
+                    break;
+                }
+            }
+            journeyState.originalArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+            for (int i = matchingItinerary.legs.size() - 1; i >= 0; i--) {
+                Leg leg = matchingItinerary.legs.get(i);
+                if (leg.transitLeg) {
+                    journeyState.originalArrivalTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                        leg.arrivalDelay,
+                        TimeUnit.SECONDS
+                    );
+                    break;
+                }
+            }
```

Comment:
Feels like there is an opportunity to encapsulate this into a method: `public static long getTransitStartOrEnd(boolean start)`. Having two back-to-back for loops like this kind of breaks the flow when reading.

Also, does this only need to happen once? I'm having trouble understanding why this gets executed each time the trip is checked.

Reply:
This isn't executed each time a trip is checked since it only occurs if there isn't yet a matching itinerary or if the matching itinerary is no longer active or upcoming. I went ahead an created some methods in the Itinerary class in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14